presto-main/src/main/java/com/facebook/presto/tdigest/TDigestUtils.java

```diff
@@ -18,12 +32,105 @@
 package com.facebook.presto.tdigest;
 
 import java.util.Random;
+import java.util.concurrent.ThreadLocalRandom;
 
-/**
- * Static sorting methods
- */
-public class Sort {
-    private static final Random prng = new Random(); // for choosing pivots during quicksort
+import static com.google.common.base.Preconditions.checkArgument;
+
+public class TDigestUtils
+{
+    private TDigestUtils() {}
+
+    static double weightedAverage(double x1, double w1, double x2, double w2)
+    {
+        if (x1 <= x2) {
+            return weightedAverageSorted(x1, w1, x2, w2);
+        }
+        else {
+            return weightedAverageSorted(x2, w2, x1, w1);
+        }
+    }
+
+    private static double weightedAverageSorted(double x1, double w1, double x2, double w2)
```

Comment:
Is this part of some existing code? Consider the 2 line logic in this I feel it makes more sense to have this to be part of the weightedAverage function
```
if (x1 > x2) {
    swap(x1, x2);
}
double x = (x1 * w1 + x2 * w2) / (w1 + w2);
return Math.max(x1, Math.min(x, x2));
```

Reply:
You can't really build a function to swap doubles, unless you pass in Double objects. I could swap them on the spot, but you would have to write 5 lines of code, since you also need to swap the weights. So it'd look something like this:
```
double temp = x1;
x1 = x2;
x2 = temp;
temp = w1;
w1 = w2;
w2 = temp;
```
I feel like the way it is right now is cleaner. I also dropped the `checkArgument` call, since it's a private function and we know from the comparison in the previous function `if (x1 > x2)` that the arguments will be passed in the correct order.